packages/datadog-plugin-mocha/src/index.js

```diff
@@ -0,0 +1,141 @@
+const { promisify } = require('util')
+
+const id = require('../../dd-trace/src/id')
+const { SAMPLING_RULE_DECISION } = require('../../dd-trace/src/constants')
+const { SAMPLING_PRIORITY } = require('../../../ext/tags')
+const { AUTO_KEEP } = require('../../../ext/priority')
+const { getGitMetadata } = require('../../dd-trace/src/plugins/util/git')
+const { getCIMetadata } = require('../../dd-trace/src/plugins/util/ci')
+const {
+  TEST_FRAMEWORK,
+  TEST_TYPE,
+  TEST_NAME,
+  TEST_SUITE,
+  TEST_STATUS
+} = require('../../dd-trace/src/plugins/util/test')
+
+const SPAN_TYPE = 'span.type'
+const RESOURCE_NAME = 'resource.name'
+
+function getTestMetadata () {
+  // TODO: eventually these will come from the tracer (generally available)
+  const ciMetadata = getCIMetadata()
+  const gitMetadata = getGitMetadata()
+
+  return {
+    [TEST_FRAMEWORK]: 'mocha',
+    ...ciMetadata,
+    ...gitMetadata
+  }
+}
+
+function createWrapRunTest (tracer, testMetadata) {
+  return function wrapRunTest (runTest) {
+    return async function runTestWithTrace () {
+      const childOf = tracer.extract('text_map', {
+        'x-datadog-trace-id': id().toString(10),
+        'x-datadog-parent-id': '0000000000000000',
+        'x-datadog-sampled': 1
+      })
+      const { file: testSuite, title: testName } = this.test
+      let specFunction = this.test.fn
+
+      if (specFunction.length) {
+        specFunction = promisify(specFunction)
+        // otherwise you have to explicitly call done()
+        this.test.async = 0
```

Comment:
Isn't this changing the behaviour of the test?

Reply:
Not that I've seen with my tests. Plus this seems to only modify the way the spec function is called (not the spec function itself), which we do anyway by `promisify`ing it:
https://github.com/mochajs/mocha/blob/c6856babc32b8556302a697429114d917dc70427/lib/runnable.js#L326-L352